### ISSUE TITLE
Design the new user experience for primary/declared data summaries in SCTK

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-toolkit#2842]
We have many related options and we want to design a new, better approach to get primary, declared licensing and package data. What would it be?

This is the current set of options:
```
    --consolidate            Group resources by Packages or license and
                             copyright holder and return those groupings as a
                             list of consolidated packages and a list of
                             consolidated components. This requires the scan to
                             have/be run with the copyright, license, and
                             package options active
    --license-clarity-score  Compute a summary license clarity score at the
                             codebase level.
    --license-policy FILE    Load a License Policy file and apply it to the scan
                             at the Resource level.
    --summary                Summarize license, copyright and other scans at the
                             codebase level.
    --summary-by-facet       Summarize license, copyright and other scans and
                             group the results by facet.
    --summary-key-files      Summarize license, copyright and other scans for
                             key, top-level files. Key files are top-level
                             codebase files such as COPYING, README and package
                             manifests as reported by the --classify option
                             "is_legal", "is_readme", "is_manifest" and
                             "is_top_level" flags.
    --summary-with-details   Summarize license, copyright and other scans at the
                             codebase level, keeping intermediate details at the
                             file and directory level.
```

**Comment by pombredanne** [comment]
Here is consensus from some chat and discussions: deprecate and remove in the version after next all these options
- consolidate
- license-clarity-score
- summary
- summary-by-facet
- summary-key-files
- summary-with-details

The new "primary", may be reusing code from summary-key-files is tracked in https://github.com/nexB/scancode-toolkit/issues/2065 and will be always ON in the default JSON/YAML output.

The new license clarity scoring is tracked in https://github.com/nexB/scancode-toolkit/issues/2861 and will be always ON in the default JSON/YAML output.

So the key is clarity and primary becomes standard, non opt-in, always on options

**Comment by pombredanne** [comment]
And the `--classify` option should become default too implicitly to support the above

**Comment by pombredanne** [comment]
We now have a new `--summary2` option by @JonoYang that gets a decent primary license for a whole codebase. What we do not have yet is the surfacing the primary license in a package.

**Comment by pombredanne** [comment]
The latest approach is going to keep the old --summary and rename it as --tallies and the new summary is replacing it and  --consolidate will be deprecated.

**Comment by JonoYang** [comment]
We have updated the summary plugin with the new fields and scoring critera and merged the code in #2914

**Comment by AyanSinhaMahapatra** [comment]
Btw, we have not made the options default (--classify/--summary/--license-clarity-score), yet. This is for the next release tentatively?
